quisp/test_utils/ModuleType.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <omnetpp.h>
+#include <string>
+#include "Configuration.h"
+
+namespace quisp_test {
+namespace module_type {
+
+using namespace omnetpp;
+
+class TestModuleType : public cModuleType {
+ private:
+  cProperties *props;
+  cConfiguration *cfg;
+
+ public:
+  TestModuleType(const char *name) : cModuleType(name) {
+    using quisp_test::configuration::Configuration;
+    cfg = new Configuration;
+    props = new cProperties;
+  }
+
+  cModule *createModuleObject() override { return nullptr; };
+  void addParametersAndGatesTo(cModule *module) override{};
+  void applyPatternAssignments(cComponent *component) override{};
+  void setupGateVectors(cModule *module) override{};
+  void buildInside(cModule *module) override{};
+
+  // methods redefined from cComponentType
+  cProperties *getProperties() const override { return props; };
+  cProperties *getParamProperties(const char *paramName) const override { return nullptr; };
```

Comment:
Now here most of the functions return nullptr. When we actually use this, we have to properly implement the functions at .cc ?

Reply:
yeah if our components need the return value of this method, somehow we need to implement it.